railsware/tap-google-ads
# [CPl-13541] Add new report and remove incompatible metrics

[Ticket link](https://railsware.atlassian.net/browse/CPL-13541)

### Problem/domain

After discussion with @paladiy we decided to create new report `Campaign performance with uniq user stats` because of conflict

`CRITICAL Cannot select the following segments because at least one unsupported metric is found in SELECT or WHERE clause: 'segments.ad_network_type'(unsupported metrics: 'unique_users', 'sk_ad_network_installs', 'average_impression_frequency_per_user')`

Also because of conflict with `segments.keyword.ad_group_criterion` and new [shopping](https://developers.google.com/google-ads/api/docs/release-notes#shopping_2:~:text=Added%20following%20conversions%20with%20cart%20data%20metrics) fields, we desided to remove new metrics from `Ad performance` report.

### Tech changes

- Removed conflict metrics
- Added new entity and list of fields for it

### Product changes



### How to test

- Open web application
- Create or edit existing importer with `Google Ads` source
- Select `Campaign performance with uniq user stats` entity

ER:
- Importer should run successfully
- Result should have `unique_users`, `sk_ad_network_installs`, `average_impression_frequency_per_user` metricts and doesn't have `segments.ad_network_type`

NEXT:
- Select `Ad performance` entity

ER:
- Importer should run successfully
- Result should have `segments.keyword.ad_group_criterion` field and doesn't have new [shopping](https://developers.google.com/google-ads/api/docs/release-notes#shopping_2:~:text=Added%20following%20conversions%20with%20cart%20data%20metrics) metrics

### How to deploy
